### PR TITLE
EZP-26068: Adopt google maps APIs v3

### PR DESCRIPTION
**Pull request by iosolinf**
Adopted google maps api v3 script.
Implemented google maps api v3 key definition and usage
Include Gmaps JS only once if the field is used several times in the page

Introduced scrollwhell parameter (this was my specific goal, to be able to disable mouse scroll from zooming the map :-))

### CODE REVIEW

**Comment by iosolinf** [comment]
Sorry... I'm not sure what is the error now. Probably the usage of the mock method...
In any case, I'm not sure if this approach (new setting and new twig extension) is to your like...
Feedback is welcomed ;)

File: eZ/Publish/Core/MVC/Symfony/Templating/Tests/Twig/Extension/ContentExtensionTest.php [hunk 1]
```diff
@@ -48,7 +48,8 @@ public function getExtensions()
                     array(),
                     $this->getMock('Psr\Log\LoggerInterface')
                 ),
-                $this->fieldHelperMock
+                $this->fieldHelperMock,
```
Comment (alongosz) [inline eZ/Publish/Core/MVC/Symfony/Templating/Tests/Twig/Extension/ContentExtensionTest.php]: @iosolinf: you just have a trailing whitespace at the end of this line :) You can use [PHP Coding Standards Fixer](http://cs.sensiolabs.org/) v1.11.x with config file `.php_cs` provided in this repository to fix any CS issues in php files.

**Comment by iosolinf** [comment]
Thank you, @alongosz 
Fixed the CS.

**Comment by bdunogier** [comment]
I'm not sure I'd use a full-featured twig helper for this. What about using the `ezpublish.configResolver` one instead ? `ezpublish.configResolver.parameter('google_maps_api_key')` ?

Furthermore, I'm thinking that this parameter is maybe in a bit of a generic place. If we add this kind of parameter at this level, this section is gonna end up really messy.

**Comment by iosolinf** [comment]
Thank you for the input, @bdunogier 
I was not comfortable with the helper function myself. I'll replace in a couple of days.
Regarding the parameter section, I am open to any suggestions or specs on where to put it.

**Comment by andrerom** [comment]
@bdunogier Any clarification on what section it should rather be in? Any context here?

**Comment by iosolinf** [comment]
Hi. I am probably not setting my environment as I should, otherwise I would not be running into the current issue
I've installed ezplatform using composer install. then I backed up ezpublish-kernel folder and replaced it with a symlink to a clone of my fork of this repository. Suddenly I am no longer able to open the front end on my environment (Error: Undefined class constant 'DISALLOW_LINKS')
I believe that oneup/flysystem-bundle needs to be updated to 1.0 here (in the sequence of https://github.com/ezsystems/ezpublish-kernel/pull/1695/files). I verify that this commit is properly merged in my environment, but running composer update on ezplatform root folder does not update flysystem...
Question: How should I setup my environment in order to avoid situations such as these?
Is there a way to run composer update so that changes to ezpublish-kernel requirements are taken into consideration?
Cheers & thanks
Io

**Comment by iosolinf** [comment]
@bdunogier {{ezpublish...}} does not seem to be available in content_fields.html.twig , that was why I developed the helper function (which I also agree is overkill and ill placed)
Any further suggestions on how to retrieve the parameter and where to place it under config?

**Comment by iosolinf** [comment]
ping @bdunogier , Can I have you input on this, in order to progress the PR? Thanks.

**Comment by bdunogier2** [comment]
> {ezpublish...}} does not seem to be available in content_fields.html.twig , that was why I developed the helper function (which I also agree is overkill and ill placed)

Odd, as this helper is a global one. I'll give it a try myself.

**Comment by bdunogier2** [comment]
See https://github.com/ezsystems/ezpublish-kernel/pull/1755.

**Comment by bdunogier2** [comment]
Please rebase your PR onto master to get rid of the conflicts. #1755 will have to be merged first.

**Comment by andrerom** [comment]
Replaces by #1820

**Pull request closed without merging**